src/praxis/momentum-root-mean-squared-propagation.ts

```diff
@@ -0,0 +1,175 @@
+import { BasePraxis, ILayerTemplate, IPraxisSettings } from './base-praxis';
+
+import { makeKernelMap, release } from '../utilities/kernel';
+import { zeros2D } from '../utilities/zeros-2d';
+import {
+  IConstantsThis,
+  IKernelFunctionThis,
+  IKernelMapRunShortcut,
+  ISubKernelObject,
+  KernelOutput,
+} from 'gpu.js';
+import { ILayer } from '../layer/base-layer';
+
+export function getMomentum(
+  delta: number,
+  decay: number,
+  previousMomentum: number
+): number {
+  return previousMomentum * decay + (1 - decay) * delta * delta;
+}
+
+export function clipByValue(value: number, max: number, min: number): number {
+  if (value > max) {
+    return max;
+  }
+  if (value < min) {
+    return min;
+  }
+  return value;
+}
+
+interface IUpdate extends IConstantsThis {
+  clipValue: number;
+  decayRate: number;
+  smoothEps: number;
+  regularizationStrength: number;
+}
+/**
+ * @description Momentum Root Mean Square Propagation Function
+ */
+export function update(
+  this: IKernelFunctionThis<IUpdate>,
+  weights: number[][],
+  deltas: number[][],
+  previousMomenta: number[][]
+): number {
+  const delta = deltas[this.thread.y][this.thread.x];
+  const clippedDelta = clipByValue(
+    delta,
+    this.constants.clipValue,
+    -this.constants.clipValue
+  );
+  const weight = weights[this.thread.y][this.thread.x];
+  const previousMomentum = previousMomenta[this.thread.y][this.thread.x];
+  const momentum = getMomentum(
+    delta,
+    this.constants.decayRate,
+    previousMomentum
+  );
+  return (
+    weight +
+    (-this.constants.learningRate * clippedDelta) /
+      Math.sqrt(momentum + this.constants.smoothEps) -
+    this.constants.regularizationStrength * weight
+  );
+}
+
+export function isClippedByValue(
+  value: number,
+  max: number,
+  min: number
+): number {
```

Comment:
we can reverse the logic to simplify code and also use `boolean` matching the function naming.